### PR TITLE
Error detection at File sharing sidebar

### LINKED ISSUE #5396: [a11y] 9.3.3.1 - Error detection
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------
## Status: partly fulfilled
- **Page 1 (Alle Dateien)**
When creating a public link and in case the user leaves the fields empty, an error message appears above the form:
**a)** The error message is in English language but should be in German - "error creating public share"
**b)** The text of the error message is not informative because it doesn’t explain what went wrong (what data caused the error?)
**c)** The error message is not handled by screenreaders. One solution could be the use of aria- live or role="alert“.

## Status: fulfilled
- **Page 2 (Mit anderen geteilten Dateien)**

## Status: not applicable
- **Page 3 (Gelöschte Dateien)**
- **Page 4 (Konto)**

### PR DESCRIPTION
**Pull request by paulcod3**
## Description
See #5396 and my comment

## Related Issue

- Fixes #5396 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingBasic</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderPermissions</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingBasic</strong> failed. The build is cancelled...</summary>

**Comment by paulcod3** [comment]
@fschade added translation for errors, could be reviewed

**Comment by paulcod3** [comment]
Translation of that error still doesnt seem to work :/ dont merge yet

**Comment by paulcod3** [comment]
Should discuss how to handle the translation for browser errors before merging

File: packages/web-app-files/src/components/PublicLinksSidebar/EditPublicLink.vue
Comment (pascalwengerter) [inline packages/web-app-files/src/components/PublicLinksSidebar/EditPublicLink.vue:10]: @paulcod3 do you remember the outcome of our discussion about translating error messages? 🤔 I don't anymore, sadly

**Review by pascalwengerter — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/components/FileSharingSidebar.vue
Comment (paulcod3) [reply packages/web-app-files/src/components/FileSharingSidebar.vue:38]: im not sure, it says "The error message is not handled by screenreaders. One solution could be the use of aria- live or role="alert“."

Comment (paulcod3) [reply packages/web-app-files/src/components/FileSharingSidebar.vue:38]: I don't know if an OcAlert would fit there

Comment (pascalwengerter) [reply packages/web-app-files/src/components/FileSharingSidebar.vue:38]: `LinkEdit` and `EditCollaborator` use the `<oc-alert>` component from ODS to display errors, and this is the element which gets appended to the DOM and needs to be communicated to the screenreader (e.g. by adding the `role="alert"` from my understanding

Comment (pascalwengerter) [reply packages/web-app-files/src/components/FileSharingSidebar.vue:38]: I'll revert/overwrite your changes, don't worry

File: packages/web-app-files/src/components/PublicLinksSidebar/EditPublicLink.vue
Comment (kulmann) [reply packages/web-app-files/src/components/PublicLinksSidebar/EditPublicLink.vue:10]: This doesn't help with translations, because we also need all possible error messages in transifex / in the translations template first.

**Comment by pascalwengerter** [comment]
Partly fixed by https://github.com/owncloud/owncloud-design-system/commit/895c0908c44b6fa33ae6fddfc0ec4bde99b9895c, closing this PR since we'll have to come up with a different solution for translating error messages

**Pull request closed without merging**
